lib/compat/wordpress-6.5/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,672 @@
+				$style_attribute_value     = $p->get_attribute( 'style' );
+				$style_attribute_value     = ( $style_attribute_value && ! is_bool( $style_attribute_value ) ) ? $style_attribute_value : '';
+
+				/*
+				 * Checks first if the style property is not falsy and the style
+				 * attribute value is not empty because if it is, it doesn't need to
+				 * update the attribute value.
+				 */
+				if ( $style_property_value || ( ! $style_property_value && $style_attribute_value ) ) {
+					$style_attribute_value = $this->set_style_property( $style_attribute_value, $style_property, $style_property_value );
+					/*
+					 * If the style attribute value is not empty, it sets it. Otherwise,
+					 * it removes it.
+					 */
+					if ( ! empty( $style_attribute_value ) ) {
+						$p->set_attribute( 'style', $style_attribute_value );
+					} else {
+						$p->remove_attribute( 'style' );
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sets an individual style property in the `style` attribute of an HTML
+	 * element, updating or removing the property when necessary.
+	 *
+	 * @since 6.5.0
+	 *
+	 * Example:
+	 *
+	 *     set_style_property( 'color:green;', 'color', 'red' )      => 'color:red;'
+	 *     set_style_property( 'background:green;', 'color', 'red' ) => 'background:green;color:red;'
+	 *     set_style_property( 'color:green;', 'color', null )       => ''
+	 *
+	 * @param string            $style_attribute_value The current style attribute value.
+	 * @param string            $style_property_name   The style property name to set.
+	 * @param string|false|null $style_property_value  The value to set for the style property. With false, null or an
+	 *                                                 empty string, it removes the style property.
+	 * @return string The new style attribute value after the specified property has been added, updated or removed.
+	 */
+	private function set_style_property( string $style_attribute_value, string $style_property_name, $style_property_value ): string {
+		$style_assignments    = explode( ';', $style_attribute_value );
+		$result               = array();
+		$style_property_found = false;
+		$style_property_value = ! empty( $style_property_value ) ? rtrim( trim( $style_property_value ), ';' ) : null;
+
+		// Searches for the style property in the existing properties.
+		foreach ( $style_assignments as $style_assignment ) {
+			if ( empty( trim( $style_assignment ) ) ) {
+				continue;
+			}
+			list( $name, $value ) = explode( ':', $style_assignment );
+			if ( trim( $name ) === $style_property_name ) {
+				// If the value is false, it omits it to the result.
+				if ( $style_property_value ) {
+					$result[] = $style_property_name . ':' . $style_property_value . ';';
+				}
+				$style_property_found = true;
+			} else {
+				$result[] = trim( $name ) . ':' . trim( $value ) . ';';
+			}
+		}
+
+		// If it doesn't find the style property, it adds it at the end of the list.
```

Comment:
there's a distinction in behavior here that might be worth noting in the documentation: appending a style property to the end of the inline style may produce a different result than replacing one that's already there.

consider properties with multiple properties that can change the same thing, like border, padding, margin, etc…

```html
<span style="border-style: dashed; border: 3px solid red;"></span>

Replace existing style.
<span style="border-style: inset; border: 3px solid red;"></span>

Append to end.
<span style="border-style: dashed; border: 3px solid red; border-style: inset;"></span>
```

In this example, replacement has left the property unaffected while appending has set it to most closely match the intention of this function call, assuming this function call wants to force the border-style value.

Without a full inline CSS parser (not just the lexical tokens, but also the semantics of constructing the full property set) it gets tricky making tactical inline CSS edits like this.

Reply:
Good point. I've moved the modified properties to the end in https://github.com/WordPress/gutenberg/pull/58066/commits/8ef546f276b6583a6ec25a7c41dc02d9358beaa2, just in case.